File: OfflineFetcher/app/src/main/java/de/jbamberger/offlinefetcher/di/reddit/RedditModule.java
package de.jbamberger.offlinefetcher.di.reddit;


import android.app.Activity;

import dagger.Binds;
import dagger.Module;
import dagger.android.ActivityKey;
import dagger.android.AndroidInjector;
import dagger.multibindings.IntoMap;
import de.jbamberger.offlinefetcher.ui.reddit.RedditActivity;

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

@Module
public abstract class RedditModule {

    @Binds
    @IntoMap
    @ActivityKey(RedditActivity.class)
    abstract AndroidInjector.Factory<? extends Activity>
    bindRedditActivityInjectorFactory(RedditSubComponent.Builder builder);
}

File: OfflineClient/app/src/main/java/de/jbamberger/offlineclient/di/reddit/RedditModule.kt
package de.jbamberger.offlineclient.di.reddit


import android.app.Activity

import dagger.Binds
import dagger.Module
import dagger.android.ActivityKey
import dagger.android.AndroidInjector
import dagger.multibindings.IntoMap
import de.jbamberger.offlineclient.ui.reddit.RedditActivity

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

@Module
abstract class RedditModule {

    @Binds
    @IntoMap
    @ActivityKey(RedditActivity::class)
    internal abstract fun bindRedditActivityInjectorFactory(builder: RedditSubComponent.Builder): AndroidInjector.Factory<out Activity>
}
